packages/tests/server/createProcedureExtension.test.ts

```diff
@@ -0,0 +1,534 @@
+import {
+  ProcedureBuilder,
+  createProcedureExtension,
+  initTRPC,
+  withPrequisites,
+} from '@trpc/server/src';
+import { z } from 'zod';
+
+type NoAny<T> = any extends T ? never : T;
+
+type ExactType<TActual, TExpected> = NoAny<TActual> extends NoAny<TExpected>
+  ? NoAny<TExpected> extends NoAny<TActual>
+    ? TActual
+    : never
+  : never;
+
+function assertType<TExpected>() {
+  return {
+    is<TActual>(_t: ExactType<TActual, TExpected>) {
+      //
+    },
+  };
+}
+
+test('complete example', async () => {
+  const t = initTRPC.context<{ userId: string }>().create();
+
+  const extension = createProcedureExtension((proc) => {
+    return proc.input(z.object({ orgId: z.number() })).use(({ next }) => {
+      return next({
+        ctx: {
+          orgPermitted: true,
+        },
+      });
+    });
+  });
+
+  const subject = t.procedure
+    .input(z.object({ name: z.string() }))
+    .extend(extension)
+    .input(z.object({ name2: z.string() }))
+    .query((opts) => {
+      assertType<string>().is(opts.input.name);
```

Comment:
`expectTypeOf(opts.input.name).toEqual<string>`

Reply:
Wow that will do!